src/main/java/net/pms/dlna/ImageInfo.java

```diff
@@ -0,0 +1,621 @@
+						height = ((IcoDirectory) directory).getInt(IcoDirectory.TAG_IMAGE_HEIGHT);
+					}
+				} else if (directory instanceof JfifDirectory) {
+					if (
+						((JfifDirectory) directory).containsTag(JfifDirectory.TAG_RESX) &&
+						((JfifDirectory) directory).containsTag(JfifDirectory.TAG_RESY)
+					) {
+						width = ((JfifDirectory) directory).getInt(JfifDirectory.TAG_RESX);
+						height = ((JfifDirectory) directory).getInt(JfifDirectory.TAG_RESY);
+					}
+				} else if (directory instanceof PcxDirectory) {
+					if (
+						((PcxDirectory) directory).containsTag(PcxDirectory.TAG_XMIN) &&
+						((PcxDirectory) directory).containsTag(PcxDirectory.TAG_XMAX) &&
+						((PcxDirectory) directory).containsTag(PcxDirectory.TAG_YMIN) &&
+						((PcxDirectory) directory).containsTag(PcxDirectory.TAG_YMAX)
+					) {
+						width = ((PcxDirectory) directory).getInt(PcxDirectory.TAG_XMAX) - ((PcxDirectory) directory).getInt(PcxDirectory.TAG_XMIN) + 1;
+						height = ((PcxDirectory) directory).getInt(PcxDirectory.TAG_YMAX) - ((PcxDirectory) directory).getInt(PcxDirectory.TAG_YMIN) + 1;
+					}
+				} else if (directory instanceof PsdHeaderDirectory) {
+					if (
+						((PsdHeaderDirectory) directory).containsTag(PsdHeaderDirectory.TAG_IMAGE_WIDTH) &&
+						((PsdHeaderDirectory) directory).containsTag(PsdHeaderDirectory.TAG_IMAGE_HEIGHT)
+					) {
+						width = ((PsdHeaderDirectory) directory).getInt(PsdHeaderDirectory.TAG_IMAGE_WIDTH);
+						height = ((PsdHeaderDirectory) directory).getInt(PsdHeaderDirectory.TAG_IMAGE_HEIGHT);
+					}
+					if (((PsdHeaderDirectory) directory).containsTag(PsdHeaderDirectory.TAG_BITS_PER_CHANNEL)) {
+						bitDepth = ((PsdHeaderDirectory) directory).getInt(PsdHeaderDirectory.TAG_BITS_PER_CHANNEL);
+					}
+					if (((PsdHeaderDirectory) directory).containsTag(PsdHeaderDirectory.TAG_COLOR_MODE)) {
+						/*
+					     * Bitmap = 0; Grayscale = 1; Indexed = 2; RGB = 3; CMYK = 4; Multichannel = 7; Duotone = 8; Lab = 9.
+						 */
+						switch (((PsdHeaderDirectory) directory).getInt(PsdHeaderDirectory.TAG_COLOR_MODE)) {
+							case 1:
+								numComponents = 1;
+								colorSpaceType = ColorSpaceType.TYPE_GRAY;
+								break;
+							case 2:
+							case 3:
+								numComponents = 3;
+								colorSpaceType = ColorSpaceType.TYPE_RGB;
+								break;
+							case 4:
+								numComponents = 4;
+								colorSpaceType = ColorSpaceType.TYPE_CMYK;
+								break;
+							default:
+						}
+					}
+
+				} else if (directory instanceof WebpDirectory) {
+					if (
+						((WebpDirectory) directory).containsTag(WebpDirectory.TAG_IMAGE_WIDTH) &&
+						((WebpDirectory) directory).containsTag(WebpDirectory.TAG_IMAGE_HEIGHT)
+					) {
+						width = ((WebpDirectory) directory).getInt(WebpDirectory.TAG_IMAGE_WIDTH);
+						height = ((WebpDirectory) directory).getInt(WebpDirectory.TAG_IMAGE_HEIGHT);
+					}
+				}
+			}
+		}
+		if (throwOnParseFailure && (width < 0 || height < 0)) {
```

Comment:
@Sami32 Zero width or height is a strange property for an image, but I use ```-1``` for "unknown"/```null``` for ```int``` since primary data types can't be ```null```. That's what I'm testing for there.